src/Testing/src/xunit/AspNetTestAssemblyRunner.cs

```diff
@@ -33,24 +34,39 @@ protected override async Task AfterTestAssemblyStartingAsync()
         // Find all the AssemblyFixtureAttributes on the test assembly
         await Aggregator.RunAsync(async () =>
         {
-            var fixturesAttributes = ((IReflectionAssemblyInfo)TestAssembly.Assembly)
-                .Assembly
+            var assembly = ((IReflectionAssemblyInfo)TestAssembly.Assembly).Assembly;
+            var fixturesAttributes = assembly
                 .GetCustomAttributes(typeof(AssemblyFixtureAttribute), false)
                 .Cast<AssemblyFixtureAttribute>()
                 .ToList();
 
             // Instantiate all the fixtures
             foreach (var fixtureAttribute in fixturesAttributes)
             {
-                var ctorWithDiagnostics = fixtureAttribute.FixtureType.GetConstructor(new[] { typeof(IMessageSink) });
                 object instance = null;
-                if (ctorWithDiagnostics != null)
+                var staticCreator = fixtureAttribute.FixtureType.GetMethod(
+                    name: "ForAssembly",
+                    bindingAttr: BindingFlags.Public | BindingFlags.Static,
+                    binder: null,
+                    types: new[] { typeof(Assembly) },
+                    modifiers: null);
```

Comment:
This is specific to `AssemblyTestLog` currently but is a pattern other classes could follow. Adding a default constructor to `AssemblyTestLog` wouldn't be a good idea.